File: src/main/java/fr/traqueur/energylib/UpdaterNetworksTask.java
package fr.traqueur.energylib;

import fr.traqueur.energylib.api.EnergyManager;

public class UpdaterNetworksTask implements Runnable {

    private final EnergyManager manager;

    public UpdaterNetworksTask(EnergyManager manager) {
        this.manager = manager;
    }

    @Override
    public void run() {
        manager.getNetworks().forEach(energyNetwork -> {
            energyNetwork.setEnable(energyNetwork.getChunk().isLoaded());
            if (energyNetwork.isEnable()) {
                energyNetwork.update();
            }
        });
    }
}

File: src/main/java/fr/traqueur/energylib/UpdaterNetworksTask.kt
package fr.traqueur.energylib

import fr.traqueur.energylib.api.EnergyManager

class UpdaterNetworksTask(private val manager: EnergyManager) : Runnable {
    override fun run() {
        manager.networks!!.forEach({ energyNetwork ->
            if (energyNetwork!!.chunk?.isLoaded == true) {
                energyNetwork.update()
            }
        })
    }
}
